### PR TITLE
Viresh_Assignment_17_Done_10+40Mins

### PR DESCRIPTION
**Pull request by VireshJJun20**
Please review and comment.

image
image

### CODE REVIEW

File: src/vireshJJun20/Assignment17/IndexCheck.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+		}
+	}
+
+	void checkIndex1(String str) { // This method gives the expected output, but not sure about the logic if its good.
+		
+		for(int i=0; i<str.length();i++) {
+			char ch1= str.charAt(i);
+			int ch1Cnt=str.indexOf(ch1);
+			//int cnt=0;
+			if (i== str.indexOf(ch1)) {
```
Comment (KrishnaKTechnocredits) [inline src/vireshJJun20/Assignment17/IndexCheck.java]: Here one more and condition required with if block that is str.indexOf(ch) != str.lastIndexOf(ch)
Try this and commit code.

File: src/vireshJJun20/Assignment17/StringMembers.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+	void CheckString(String str){
+		int cLetter=0;
+		int cLower=0;
+		int cUpper=0;
+		int cDigit=0;
+		int cOther=0;
+		for (int i=0; i<str.length(); i++){
+			if(Character.isLetter(str.charAt(i))){
+				cLetter++;
```
Comment (KrishnaKTechnocredits) [inline src/vireshJJun20/Assignment17/StringMembers.java:13]: Are you. sure this if block is required. ? cLetter without using this variable you can print expected output.

File: src/vireshJJun20/Assignment17/IndexCheck.java (hunk 1 above)
Comment (VireshJJun20) [reply src/vireshJJun20/Assignment17/IndexCheck.java]: Yes.. that was missing, I had added it and it worked perfectly. Thanks Krishna.

image

File: src/vireshJJun20/Assignment17/StringMembers.java (hunk 2 above)
Comment (VireshJJun20) [reply src/vireshJJun20/Assignment17/StringMembers.java:13]: I got your point !! (I added that if block to explicitely include isLetter() method.. ) 
Thanks Krishna. 
Updated my code and now it look like below..
image

Comment (KrishnaKTechnocredits) [reply src/vireshJJun20/Assignment17/StringMembers.java:13]: Now it's perfect code.

**Review by KrishnaKTechnocredits — APPROVED** [review]

**Pull request merged**